File: Oops Notes/NOTES FOR STUDIES/Abdul rub/VB.NET/DatabaseApp/DatabaseApp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DatabaseApp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'employeeDataSet.emp' table. You can move, or remove it, as needed.
            this.empTableAdapter.Fill(this.employeeDataSet.emp);

        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (this.empBindingSource.Position < this.empBindingSource.Count - 1)
                this.empBindingSource.MoveNext(); 
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if (this.empBindingSource.Position >0)
                this.empBindingSource.MovePrevious ();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            
                this.empBindingSource.MoveFirst ();
        }

        private void button4_Click(object sender, EventArgs e)
        {
            this.empBindingSource.MoveLast();
        }
    }
}

File: Oops Notes/NOTES FOR STUDIES/Abdul rub/VB.NET/practice/database/WindowsApplication1/WindowsApplication1/Form1.vb
Public Class Form1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'TODO: This line of code loads data into the 'EmployeeDataSet.emp' table. You can move, or remove it, as needed.
        Me.EmpTableAdapter.Fill(Me.EmployeeDataSet.emp)

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If Me.EmpBindingSource.Position < Me.EmpBindingSource.Count - 1 Then
            Me.EmpBindingSource.MoveNext()
        End If

    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        If Me.EmpBindingSource.Position > 0 Then
            Me.EmpBindingSource.MovePrevious()
        End If

    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        Me.EmpBindingSource.MoveFirst()
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        Me.EmpBindingSource.MoveLast()
    End Sub
End Class
